src/mbgl/style/expression/in.cpp

```diff
@@ -0,0 +1,118 @@
+#include <string.h>
+#include <mbgl/style/conversion_impl.hpp>
+#include <mbgl/style/expression/in.hpp>
+#include <mbgl/style/expression/type.hpp>
+#include <mbgl/util/string.hpp>
+
+namespace mbgl {
+namespace style {
+namespace expression {
+
+namespace {
+bool isComparableType(type::Type type) {
+    return type == type::Boolean || type == type::String || type == type::Number || type == type::Null ||
+        type == type::Value;
+}
+
+bool isComparableRuntimeValue(type::Type type) {
+    return type == type::Boolean || type == type::String || type == type::Number || type == type::Null;
+}
+
+bool isSearchableRuntimeValue(type::Type type) {
+    return type == type::String || type.is<type::Array>() || type == type::Null;
+}
+}
+
+EvaluationResult In::evaluate(const EvaluationContext& params) const {
+    const EvaluationResult evaluatedNeedle = needle->evaluate(params);
+    if (!evaluatedNeedle) {
+        return evaluatedNeedle.error();
+    }
+
+    const EvaluationResult evaluatedHaystack = haystack->evaluate(params);
```

Comment:
If haystack may be a nullset, it is better to evaluate it first and then `return EvaluationResult(false)` right away.